lib/rack/request.rb

```diff
@@ -16,7 +16,20 @@ class << self
       attr_accessor :ip_filter
     end
 
-    self.ip_filter = lambda { |ip| /\A127\.0\.0\.1\Z|\A(10|172\.(1[6-9]|2[0-9]|30|31)|192\.168)\.|\A::1\Z|\Afd[0-9a-f]{2}:.+|\Alocalhost\Z|\Aunix\Z|\Aunix:/i.match?(ip) }
+    valid_ipv4_octet = '(25[0-5]|2[0-4][0-9]|[01]?[0-9]?[0-9])'.freeze
```

Comment:
Looking at everywhere that `valid_ipv4_octet` is used, it is preceded by a `.`.  So it seems better to do:

```ruby
# no need to freeze this local variable
# Also makes sense to use a regexp instead of a string
valid_ipv4_octet = /\.(25[0-5]|2[0-4][0-9]|[01]?[0-9]?[0-9])/
```

That will allow you to simplify usage to:

```ruby
/\A127#{valid_ipv4_octet}{3}\z/, 
```

With similar changes to the other IPv4 regexps.

Reply:
Oh, nice catch, thank you.  That makes everything much more readable.